test/Ocelot.UnitTests/Configuration/UpstreamHeaderRoutingOptionsCreatorTests.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Linq;
+using Ocelot.Configuration.File;
+using Ocelot.Configuration.Creator;
+using Ocelot.Configuration;
+using Xunit;
+using TestStack.BDDfy;
+using Shouldly;
+
+namespace Ocelot.UnitTests.Configuration;
+
+public class UpstreamHeaderRoutingOptionsCreatorTests
+{
+    private FileUpstreamHeaderRoutingOptions _fileUpstreamHeaderRoutingOptions;
```

Comment:
It cannot. I adopted the approach from other tests and use the When.../Then... approach, so I am referencing the file routing options in `WhenICreate()`.

Reply:
I guess, It Can! Every `Given`, `And`, `When`, `Then` methods **can** have input and even output parameters!
I will show you later...

Reply:
Can you show me a practical sample of what you had in mind? Because like I said below, I understand the concept, but actually implementing it in code does not make it look feasible unless I have to take a different approach to how I then call the sequence of actions to run the test.

Reply:
Will show you after conflicts resolution, and ping me back plz.